modules/common/secret/secret.go

```diff
@@ -385,3 +386,70 @@ func GetDataFromSecret(
 
 	return data, ctrl.Result{}, nil
 }
+
+// EnsureSecretData - ensures that the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func EnsureSecretData(
+	ctx context.Context,
+	secretName types.NamespacedName,
+	expectedFields []string,
+	reader client.Reader,
+	Updater condition.Updater,
+	requeueTimeout time.Duration,
+) (string, ctrl.Result, error) {
+	secret := &corev1.Secret{}
+	err := reader.Get(ctx, secretName, secret)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// TODO(gibi): Change the message to state which input
+			// (i.e. Secret with a given name) is missing
```

Comment:
Lets resolve this TODO. I have the resolution in nova as part of a longer draft series.  https://github.com/openstack-k8s-operators/nova-operator/pull/259/commits/a6d3d2df1a278a28d293896ce40bbb87b5afe780#diff-da55647121285e1a60376c19fef7ce508bd29edddabac15bc6c8781be47565ee 

However it needs to change condition.InputReadyWaitingMessage to allow incorporating the missing input to the message. And that is a breaking change if done in lib-common

Reply:
this does not really match or naming convention by the way.

we use ensure to mean validate and create/update if not correct.
this is a bug in the nova operator naming of this function.

this is actually VerifySecret

since this checks that the secret exists and that the required field are present but it will never ensure the value are set to a specific value.

can we rename it to VerifySecret

Reply:
It is fine by me to rename it.

Reply:
agree with rename it to VerifySecret